evm/vm/forks/sharding/blocks.py

```diff
@@ -0,0 +1,21 @@
+from rlp.sedes import (
+    CountableList,
+)
+from evm.rlp.headers import (
+    BlockHeader,
+)
+from evm.vm.forks.homestead.blocks import (
+    HomesteadBlock,
+)
+from .transactions import (
+    ShardingTransaction,
+)
+
+
+class ShardingBlock(HomesteadBlock):
```

Comment:
yeah, the name will be changed later but for now I keep it this way for simplicity.

I suppose with new transaction class, there would have to be new block class?
the new block class will be part of the `ShardingVM`
```
ShardingVM = ByzantiumVM.configure(
    name='ShardingVM',
    _block_class=ShardingBlock,
)
```

Reply:
I think it was discussed but I can't recall the specifics.  

What do you think about just writing an entirely separate `Shard` class so that we aren't overloading the `block` based API's, trying to cram the concepts of collations and blocks into the same API.  While the two have a lot of similarities, I think they are going to be sufficiently different that we'll need a separate abstraction.

I have some deeper thoughts on the subject that I'll try to write-up since I'm starting to get the feeling that we will need a `Shard` class which mirrors the `Chain` class and than we'll want a `ShardVM` which mirrors the `VM` class.

Reply:
Agree that it's better to write a `Shard` class. But maybe it can be replaced later? My goal for this PR is to change Transaction related code. (sorry about the misleading `ShardingBlock`, I figured if I want to test the modification on Transaction I need to also set up `VM` and `Block` class for `Sharding`.)